vm_insnhelper.c

```diff
@@ -5565,6 +5566,7 @@ vm_sendish(
 VALUE
 rb_vm_send(rb_execution_context_t *ec, rb_control_frame_t *reg_cfp, CALL_DATA cd, ISEQ blockiseq)
 {
+    stack_check(ec);
```

Comment:
You seem to be adding extra checks to the interpreter as well, why?

Reply:
These are used for instruction fallback. You can see usage in our gen_send().

Reply:
I was worried since this could slow down the fallback calls, but maybe not too significant?

```
before: ruby 3.3.0dev (2023-09-14T15:11:32Z master 6df0927be2) +YJIT [x86_64-linux]
after: ruby 3.3.0dev (2023-09-14T19:33:53Z yjit-stack-overflow ec4cf7c1f6) +YJIT [x86_64-linux]

----------  -----------  ----------  ----------  ----------  -------------  ------------
bench       before (ms)  stddev (%)  after (ms)  stddev (%)  after 1st itr  before/after
lobsters    610.8        4.0         610.0       4.1         1.00           1.00
railsbench  1128.5       1.4         1121.7      1.4         1.00           1.01
----------  -----------  ----------  ----------  ----------  -------------  ------------
```